### PR TITLE
Added AnyLeaf G4

### PR DESCRIPTION
**Pull request by David-OConnor**
Tested on hardware.

### CODE REVIEW

**Comment by sugaarK** [comment]
You can’t have user settings in the target like the msp display port. The erls defines are only ok if it’s built in. Also the dshot commands are a no go as well. If it’s for a BNF you can make a preset but you can’t have a target with those defines

**Review by haslinghuis — APPROVED** [review]

**Comment by David-OConnor** [comment]
I've removed the DSHOT commands.

There is a dedicated port and set of pads for Vtx, and UART4 is not exposed in the normal UART section. There is no OSD chip on board, so UART4 is explicitly for digital OSD.

Thank you.

**Review by sugaarK — APPROVED** [review]

**Comment by sugaarK** [comment]
Are you affiliated with the company ?

**Comment by David-OConnor** [comment]
I am the owner.

File: configs/default/ANYL-MERCURYG4.config [hunk 1]
```diff
@@ -0,0 +1,142 @@
+# Betaflight / STM32G47X (SG47) 4.4.0 Feb 25 2023 / 03:35:00 (8d4f005) MSP API: 1.44
+
+#define USE_GYRO_SPI_ICM42688P
+#define USE_ACC_SPI_ICM42688P
+#define USE_BARO_I2C_DPS310
```
Comment (haslinghuis) [inline configs/default/ANYL-MERCURYG4.config:5]:
```suggestion
#define USE_BARO_DPS310
```

**Comment by sugaarK** [comment]
> I am the owner.

Nice you should join the bf partners program. 
the idea is we involve our talented team in the design process of new hardware to ensure it gets sold to the public actually reliable and fully functional with our release code. At the end of the day our team provides 90% of the support for companies products operation in our eco system so any thing to decrease our work In hardware support is a good thing for the team   


You can contact me at SugarK@betaflight.com to setup a MFG channel for your company

File: configs/default/ANYL-MERCURYG4.config [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+set gyro_1_bustype = SPI
+set gyro_1_spibus = 1
+set gyro_1_sensor_align = CW90
+
+
+# Barometer
+set baro_hardware = DPS310
+set baro_bustype = I2C
+set baro_i2c_device = 2
+set baro_i2c_address = 119  # 0x77
```
[the remark this answers is not in the document]
Comment (David-OConnor) [reply configs/default/ANYL-MERCURYG4.config:26]: They're no

File: configs/default/ANYL-MERCURYG4.config (hunk 1 above)
Comment (David-OConnor) [reply configs/default/ANYL-MERCURYG4.config]: Fixed

**Review by haslinghuis — DISMISSED** [review]

File: configs/default/ANYL-MERCURYG4.config [hunk 3]
```diff
@@ -0,0 +1,142 @@
+# Betaflight / STM32G47X (SG47) 4.4.0 Feb 25 2023 / 03:35:00 (8d4f005) MSP API: 1.44
+
+#define USE_GYRO_SPI_ICM42688P
+#define USE_ACC_SPI_ICM42688P
+#define USE_BARO_DPS310
+#define USE_FLASH_M25P16
+#define USE_GPS
+//TODO#define SYSTEM_HSE_MHZ 16
```
Comment (KarateBrot) [inline configs/default/ANYL-MERCURYG4.config:8]:
```suggestion
```
Is this comment needed? Looks like we could just remove it from the "official" config

Comment (David-OConnor) [reply configs/default/ANYL-MERCURYG4.config:8]: This is in line with some recent changes for an upcoming change.

**Review by KarateBrot — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

**Pull request merged**